packages/styletron-engine-atomic/src/server/server.js

```diff
@@ -176,17 +261,70 @@ function stringify(styleRules, sortedCacheKeys) {
   return result;
 }
 
-function sheetify(styleRules, sortedCacheKeys): Array<sheetT> {
+function sheetify(styleRules, sortedCacheKeys, streamingMode): Array<sheetT> {
   if (sortedCacheKeys.length === 0) {
     return [{css: "", attrs: {}}];
   }
   const sheets = [];
   sortedCacheKeys.forEach(cacheKey => {
     // omit media (cacheKey) attribute if empty
     const attrs = cacheKey === "" ? {} : {media: cacheKey};
-    sheets.push({css: styleRules[cacheKey], attrs});
+    let insertBeforeMedia =
+      streamingMode && styleRules[cacheKey].insertBeforeMedia
+        ? styleRules[cacheKey].insertBeforeMedia
+        : null;
+    if (streamingMode)
+      insertBeforeMedia = styleRules[cacheKey].insertBeforeMedia || null;
+    sheets.push({
+      ...(insertBeforeMedia && {insertBeforeMedia}),
+      css: styleRules[cacheKey].rules,
+      attrs,
+    });
   });
   return sheets;
 }
 
+function generateScript(sheet: sheetT, className: string, attrs: attrsT) {
+  if (!sheet.css.length) return "";
+
+  const content = [];
+  content.push(`
+  <script>
+  (function () {
+  `);
+
+  const selector = attrs.media
+    ? `style.${className}[media='${attrs.media}']`
+    : `style.${className}:not([media])`;
+
+  content.push(`
+    let styleElement = document.head.querySelector("${selector}");
+    if (!styleElement) {
+      styleElement = document.createElement("style");`);
+
+  for (const attr in attrs) {
+    content.push(`
+      styleElement.setAttribute("${attr}", "${attrs[attr]}");`);
+  }
+
+  if (sheet.insertBeforeMedia) {
+    content.push(`
+      const predecessor = document.querySelectorAll('[media="${sheet.insertBeforeMedia}"]')[0]
```

Comment:
Probably best to avoid const/let. Also accessing the first element of `querySelectorAll` is equivalent to `querySelector`, right?

```suggestion
      var predecessor = document.querySelector('[media="${sheet.insertBeforeMedia}"]');
```

Reply:
Yep, nice catch! I had fixed it locally afterwards but forgot to push that commit